### PR TITLE
Replace JAVA_BIN with TEST_JDK_HOME

### PR DESCRIPTION
**Pull request by renfeiw**
- expect user to export TEST_JDK_HOME instead of JAVA_BIN
- figure out where JAVA_BIN is based on JDK_VERSION
- remove JDK_HOME, replace it with TEST_JDK_HOME
- remove JAVA_BIN_TMP
- update readme

[ci skip]

Signed-off-by: Renfei Wang <renfeiw@ca.ibm.com>

### CODE REVIEW

**Comment by renfeiw** [comment]
reviewer: @llxia

**Comment by llxia** [comment]
Related to https://github.com/eclipse/openj9/issues/1058

File: test/README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Linux x86-64 cmprssptrs OpenJ9 SDK:
 
 ```
     cd openj9/test/TestConfig
-    export JAVA_BIN=/my/openj9/sdk/bin
+    export TEST_JDK_HOME=/my/openj9/sdk
```
Comment (llxia) [inline test/README.md]: `/my/openj9/jdk`

Comment (renfeiw) [reply test/README.md]: updated

**Comment by llxia** [comment]
Please fix the merge conflict.

**Comment by renfeiw** [comment]
fixed

**Comment by llxia** [comment]
Jenkins test sanity all JDK8

**Comment by llxia** [comment]
Jenkins test sanity all JDK11

**Comment by llxia** [comment]
This change has to work with PR in AdoptOpenJDK https://github.com/AdoptOpenJDK/openjdk-tests/pull/1093

PR builds in OpenJ9 cannot verify the change.

**Comment by llxia** [comment]
Since we tested in Grinder and internally, I will merge the change.
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/328/
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/337/

**Comment by jdekonin** [comment]
PR builds use a different mechanism for triggering testing, the errors in the jobs triggered were valid/real.  I have opened #5621 with what I believe is resolution.

**Pull request merged**